rust_src/src/keymap.rs

```diff
@@ -258,6 +262,58 @@ pub fn keymap_prompt(map: LispObject) -> LispObject {
     LispObject::constant_nil()
 }
 
+/// Same as `map_keymap_internal`, but traverses parent keymaps as well.
+/// AUTOLOAD indicates that autoloaded keymaps should be loaded.
+#[no_mangle]
+pub extern "C" fn map_keymap(
+    map: Lisp_Object,
+    fun: map_keymap_function_t,
+    args: Lisp_Object,
+    data: *const c_void,
+    autoload: bool,
+) {
+    let mut map = LispObject::from_raw(get_keymap(map, true, autoload));
+    while map.is_cons() {
+        if let Some(cons) = map.as_cons() {
```

Comment:
`let (keymap, map_1) = cons.as_tuple();`

This gives you car and cdr at once.

Reply:
Ah , that is really nice. I'll use that. Thanks!